api/src/main/java/org/apache/iceberg/AddedRowsScanTask.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+/**
+ * A scan task for inserted data records generated by adding a data file to the table.
+ * <p>
+ * Note that added files may have matching position deletes produced in the same snapshot.
```

Comment:
Do we need to care about the equality deletes as well?

Reply:
Eq delete doesn't apply to the data files in the same snapshot, which is written in the spec, https://iceberg.apache.org/spec/#scan-planning.

Reply:
Does this assume it only scans one snapshot? From the comment here:https://github.com/apache/iceberg/pull/4870/files#r889236453, it looks like support scan across snapshots?

Reply:
We need to consider equality deletes as well if this task can be used for multiple snapshots(net change use case).

Reply:
I updated the Javadoc. If scanning for changes across snapshots, we may have equality deletes too.